### PR TITLE
dnsdist: add inClientStartup() function

### PR DESCRIPTION
**Pull request by Habbie**
### Short description
This allows the user to, for example, print certain things during daemon startup but not while connecting a client.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/dnsdistdist/docs/reference/config.rst [hunk 1]
```diff
@@ -118,6 +118,10 @@ Control Socket, Console and Webserver
 
   :param str address: An IP address with optional port. By default, the port is 5199.
 
+.. function:: inClientStartup()
+
+  Returns true while the console client is parsing the configuration.
```
Comment (rgacogne) [inline pdns/dnsdistdist/docs/reference/config.rst]: Doesn't it return `true` also after the client is done parsing the configuration? I think we have something like `g_configurationDone` to indicate whether we are done parsing the configuration.

Comment (Habbie) [reply pdns/dnsdistdist/docs/reference/config.rst]: It returns false in the client console. I couldn't imagine a situation in which somebody could call this in the client state after the `>` prompt was shown so I did not think about it.

Comment (rgacogne) [reply pdns/dnsdistdist/docs/reference/config.rst]: You're right, after the client is done parsing the configuration all commands are passed to the server over the console, so my comment made no sense!

Comment (Habbie) [reply pdns/dnsdistdist/docs/reference/config.rst]: pushed something for this

**Comment by Habbie** [comment]
@zeha did you want any change in the logic here?

**Comment by zeha** [comment]
@Habbie I couldn't find a variant that was strictly better, so no.

**Pull request merged**